src/beacons/permissions.ts

```diff
@@ -1,99 +1,124 @@
-        );
+  if (bluetoothStatus === RESULTS.GRANTED) {
+    permissionRequiredForBeaconsGranted = true;
+    // Request location when in use
+    const locationWhenInUseStatus = await request(
+      BEACONS_PERMISSIONS.locationWhenInUse,
+      rationaleMessages.locationWhenInUse,
+    );
 
-        if (locationWhenInUseStatus === RESULTS.GRANTED) {
-            // Request location always or background location
-            const locationAlwaysStatus = await request(
-                BEACONS_PERMISSIONS.locationAlways,
-                rationaleMessages.locationAlways,
-            );
+    if (locationWhenInUseStatus === RESULTS.GRANTED) {
+      // Request location always or background location
+      const locationAlwaysStatus = await request(
+        BEACONS_PERMISSIONS.locationAlways,
+        rationaleMessages.locationAlways,
+      );
 
-            if (locationAlwaysStatus === RESULTS.GRANTED) {
-                // Request motion permission
-                await request(BEACONS_PERMISSIONS.motion, rationaleMessages.motion);
-            }
-        }
+      if (locationAlwaysStatus === RESULTS.GRANTED) {
+        // Request motion permission
+        await request(BEACONS_PERMISSIONS.motion, rationaleMessages.motion);
+      }
     }
+  }
 
-    return permissionRequiredForBeaconsGranted;
+  return permissionRequiredForBeaconsGranted;
 };
 
 const checkPermissionStatuses = async () => {
-    const statuses = await checkMultiple(Object.values(BEACONS_PERMISSIONS));
-    const permissionStatuses: Record<PermissionKey, boolean> = {
-        bluetooth: false,
-        locationWhenInUse: false,
-        locationAlways: false,
-        motion: false,
-    };
-    Object.keys(permissionStatuses).forEach((key) => {
-        const permissionKey = key as PermissionKey;
-        permissionStatuses[permissionKey] = statuses[BEACONS_PERMISSIONS[permissionKey]] === RESULTS.GRANTED;
-    });
-    return permissionStatuses;
+  const statuses = await checkMultiple(Object.values(BEACONS_PERMISSIONS));
+  const permissionStatuses: Record<PermissionKey, boolean> = {
+    bluetooth: false,
+    locationWhenInUse: false,
+    locationAlways: false,
+    motion: false,
+  };
+  Object.keys(permissionStatuses).forEach((key) => {
+    const permissionKey = key as PermissionKey;
+    permissionStatuses[permissionKey] =
+      statuses[BEACONS_PERMISSIONS[permissionKey]] === RESULTS.GRANTED;
+  });
+  return permissionStatuses;
 };
 
-export const allowedPermissionForKettle = async () => {
-    const permissionStatuses = await checkPermissionStatuses();
-    const kettleModulesArray = [];
-    if (permissionStatuses.bluetooth) {
-        kettleModulesArray.push(KettleModules.BLUETOOTH);
-    }
-    if (permissionStatuses.locationAlways) {
-        kettleModulesArray.push(KettleModules.LOCATION);
-    }
-    if (permissionStatuses.motion) {
-        kettleModulesArray.push(KettleModules.ACTIVITY);
-    }
+export const allowedPermissionForBeacons = async () => {
```

Comment:
Should be called `allowedPermissionsForBeacons` since it returns an array I think